### ISSUE TITLE
Error while creating scheduled task to cleanup blocked applications

### ISSUE THREAD

**Issue by lucafabbri365** [PSAppDeployToolkit/PSAppDeployToolkit#296]
When I use the **-BlockExecution** parameter together with _$installName_ variable set to value containing spaces, scheduled task creation (used to clean up blocked applications in case installation is interrupted) fails and the block execution doesn't work as expected (reference to: https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/issues/265).
Here the Deploy-Application.ps1 PowerShell script:

**Variable Declaration**

`## Variables: Install Titles (Only set here to override defaults set by the toolkit)`
`[string]$installName = 'My Own Script'`
`[string]$installTitle = 'My First Own Script'`

**Pre-Installation**

`[string]$installPhase = 'Pre-Installation'`
`Show-InstallationWelcome -CloseApps 'notepad' -AllowDefer -DeferTimes 3 -CheckDiskSpace -PersistPrompt`

When I launch the script manually from PowerShell console ("Run as administrator"), these rows are written into the log file:

`<![LOG[[Pre-Installation] :: Executing [C:\WINDOWS\System32\schtasks.exe /create /f /tn My Own Script_BlockedApps /xml "C:\Users\Public\PSAppDeployToolkit\SchTaskUnBlockApps.xml"]...]LOG]!>`
`<![LOG[[Pre-Backup] :: Standard error output from the process: ERROR: Invalid argument/option - 'Own'.
Type "SCHTASKS /CREATE /?" for usage`

As I can understand, to avoid issue above and let the block execution to work as expected, the _$installName_ variable must not contain any space (eg. `$installName = 'My_Own_Script'`)

Is it a known behavior ?

**Comment by SierraEcoBravo** [comment]
I believe the problem is that when the default $installName variable is created, it and several other $appxxx variables that make up $installName are scrubbed in AppDeployToolkitMain.ps1. When $installName is set in Deploy-Application.ps1, the scrubbing is bypassed.

Try adding the following after $installName in your Deploy-Application.ps1:

```
[char[]]$invalidFileNameChars = [IO.Path]::GetInvalidFileNameChars()
[string]$installName = $installName -replace "[$invalidFileNameChars]",'' -replace ' ',''
[string]$installName = $installName.Trim('_') -replace '[_]+','_'
```

**Comment by seanlillis** [comment]
Good catch. I'm pushing a fix for this now to add whitespace as an invalid character in the scheduled task creation, so it will catch whitespaces in $installname irrespective of where the variable is defined.

**Comment by seanlillis** [comment]
This is fixed in v3.7.0.
